core/src/main/java/com/digitalpebble/stormcrawler/persistence/AdaptiveScheduler.java

```diff
@@ -0,0 +1,321 @@
+
+    /**
+     * Name of key to hold previous signature: a copy, not overwritten by
+     * {@link MD5SignatureParseFilter}, is added by
+     * {@link com.digitalpebble.stormcrawler.parse.filter.SignatureCopyParseFilter}.
+     * This key is a temporary copy, not necessarily persisted in metadata.
+     */
+    public static final String SIGNATURE_OLD_KEY = "signatureOld";
+
+    /**
+     * Key to store the current fetch interval value,
+     * must be listed in &quot;metadata.persist&quot;.
+     */
+    public static final String FETCH_INTERVAL_KEY = "fetchInterval";
+
+    /**
+     * Key to store the date when the signature has been changed,
+     * must be listed in &quot;metadata.persist&quot;.
+     */
+    public static final String SIGNATURE_MODIFIED_KEY = "signatureChangeDate";
+
+
+    private static final org.slf4j.Logger LOG = LoggerFactory
+            .getLogger(AdaptiveScheduler.class);
+
+    protected int defaultfetchInterval;
+    protected int minFetchInterval = 60;
+    protected int maxFetchInterval = 60 * 24 * 14;
+    protected float fetchIntervalDecRate = .5f;
+    protected float fetchIntervalIncRate = .5f;
+
+    protected boolean setLastModified = false;
+    protected boolean overwriteLastModified = false;
+
+    private SimpleDateFormat httpDateFormat = AbstractHttpProtocol
+            .getHttpDateFormat();
+
+    @Override
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    public void init(Map stormConf) {
+        defaultfetchInterval = ConfUtils.getInt(stormConf,
+                Constants.defaultFetchIntervalParamName, 1440);
+        setLastModified = ConfUtils.getBoolean(stormConf, SET_LAST_MODIFIED,
+                false);
+        minFetchInterval = ConfUtils.getInt(stormConf, INTERVAL_MIN,
+                minFetchInterval);
+        maxFetchInterval = ConfUtils.getInt(stormConf, INTERVAL_MAX,
+                maxFetchInterval);
+        fetchIntervalDecRate = ConfUtils.getFloat(stormConf, INTERVAL_DEC_RATE,
+                fetchIntervalDecRate);
+        fetchIntervalIncRate = ConfUtils.getFloat(stormConf, INTERVAL_INC_RATE,
+                fetchIntervalIncRate);
+        super.init(stormConf);
+    }
+
+    @Override
+    public Date schedule(Status status, Metadata metadata) {
+        LOG.debug("Scheduling status: {}, metadata: {}", status, metadata);
+
+        String signature = metadata.getFirstValue(SIGNATURE_KEY);
+        String oldSignature = metadata.getFirstValue(SIGNATURE_OLD_KEY); 
+
+        if (status != Status.FETCHED) {
+
+            // reset all metadata
+            metadata.remove(SIGNATURE_MODIFIED_KEY);
+            metadata.remove(FETCH_INTERVAL_KEY);
+            metadata.remove(SIGNATURE_KEY);
+            metadata.remove(SIGNATURE_OLD_KEY);
+
+            // fall-back to DefaultScheduler
+            return super.schedule(status, metadata);
+        }
+
+        Calendar now = Calendar.getInstance(Locale.ROOT);
+
```

Comment:
what about 
```
final String now_string = httpDateFormat.format(now.getTime());
```

and use it below instead of 

`httpDateFormat.format(now.getTime())`